kong/tools/queue.lua

```diff
@@ -258,13 +257,22 @@ function Queue:process_once()
   local retry_count = 0
   while true do
     self:log_debug("passing %d entries to handler", entry_count)
-    ok, err = self.handler(self.handler_conf, {unpack(self.entries, self.front, self.front + entry_count - 1)})
-    if ok then
+    local status
+    status, ok, err = pcall(self.handler, self.handler_conf,
+                            {unpack(self.entries, self.front, self.front + entry_count - 1)})
+    if not status then
+      -- We just log the error and continue, since we want to
+      -- continue executing the remaining entries in the queue.
+      self:log_err("handler processed %d entries failed, err: %s", entry_count, ok)
```

Comment:
Instead of simply continuing as if the call succeeded, we should treat the error like one that was returned by the handler.  This will cause the call to be retried according to the retry policy, which is preferable in my opinion.

Reply:
updates

Reply:
update